### ISSUE TITLE
False positive: hosts variable not defined

### ISSUE THREAD

**Issue by skwde** [ansible/ansible-lint#3229]
##### Summary

False positve for undefined variable in hosts.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 6.10.2 using ansible 2.14.3
A new release of ansible-lint is available: 6.10.2 → 6.14.3 Upgrade by running: pip install --upgrade ansible-lint
```

- ansible installation method: python (conda)
- ansible-lint installation method: python (conda)

##### STEPS TO REPRODUCE

Save the following playbook

```console (paste below)
---
- name: Test1
  hosts: "{{ target1 }}"
  gather_facts: false
  vars_prompt:

    - name: target1
      prompt: "Specify deployment target(s)"
      private: false
      default: 'localhost'

  tasks:
    - name: Store Variable for cross play visibility
      ansible.builtin.set_fact:
        target: "{{ target1 }}"
      run_once: true
      delegate_facts: true
      delegate_to: localhost

    - name: Output1
      ansible.builtin.debug:
        msg:
          - "{{ target1 }}"
          - "{{ hostvars['localhost']['target'] }}"

- name: Test2
  hosts: "{{ hostvars['localhost']['target'] }}"
  gather_facts: true
  tasks:
    - name: Output2
      ansible.builtin.debug:
        msg:
          - "{{ hostvars['localhost']['target'] }}"
```

as `test.yml`.

Run 

```sh
ansible-lint playbooks/test1.yml
```

and get

```out
WARNING  Listing 1 violation(s) that are fatal
syntax-check[specific]: The field 'hosts' has an invalid value, which includes an undefined variable. The error was: 'hostvars' is undefined. 'hostvars' is undefined
playbooks/test2.yml:26:3

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - syntax-check[specific]  # Ansible syntax check failed.

                  Rule Violation Summary                   
 count tag                    profile rule associated tags 
     1 syntax-check[specific] min     core                 

Failed after : 1 failure(s), 0 warning(s) on 1 files.
A new release of ansible-lint is available: 6.10.2 → 6.14.3 Upgrade by running: pip install --upgrade ansible-lint
```

Note, also 

```sh
ansible-playbook --syntax-check test.yml 
```

gives

```out
[WARNING]: No inventory was parsed, only implicit localhost is available
[WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'
ERROR! The field 'hosts' has an invalid value, which includes an undefined variable. The error was: 'hostvars' is undefined. 'hostvars' is undefined

The error appears to be in '<path to test>/test.yml': line 26, column 3, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:


- name: Test2
  ^ here
```

##### Desired Behavior

No Errors from above playbook

Possible security bugs should be reported via email to `security@ansible.com`

**Comment by ssbarnea** [comment]
Please read the documentation page, it gives an exact fix for your problem and also an explanation. https://ansible-lint.readthedocs.io/rules/syntax-check/
